oh_staff_ui/management/commands/reprocess_derivative_images.py

```diff
@@ -0,0 +1,160 @@
+
+
+logger = logging.getLogger(__name__)
+
+
+def reprocess_derivative_images(master_images: QuerySet, request: HttpRequest) -> None:
+    """Reprocess all derivative images from masters."""
+    try:
+        for master_image in master_images:
+            logger.info(
+                f"Recreating derivatives for master image {master_image.file.name}."
+            )
+            master_id = master_image.id
+            # Create an OHF for the master file so we can use ImageFileHandler
+            master_file = OralHistoryFile(
+                master_image.item.id,
+                Path(settings.MEDIA_ROOT).joinpath(master_image.file.name),
+                master_image.file_type,
+                "master",
+                request,
+            )
+            handler = ImageFileHandler(master_file)
+
+            # Submaster - generate and save
+            submaster_file_name = handler.create_submaster()
+            submaster_file = OralHistoryFile(
+                item_id=master_image.item.id,
+                file_name=submaster_file_name,
+                file_type=MediaFileType.objects.get(file_code="image_submaster"),
+                file_use="submaster",
+                request=request,
+            )
+            submaster_file.process_media_file(parent_id=master_id)
+            logger.info(f"Submaster image {submaster_file_name} created.")
+
+            # Thumbnail - generate and save
+            thumbnail_file_name = handler.create_thumbnail()
+            thumbnail_file = OralHistoryFile(
+                item_id=master_image.item.id,
+                file_name=thumbnail_file_name,
+                file_type=MediaFileType.objects.get(file_code="image_thumbnail"),
+                file_use="thumbnail",
+                request=request,
+            )
+            thumbnail_file.process_media_file(parent_id=master_id)
+            logger.info(f"Thumbnail image {thumbnail_file_name} created.")
+            logger.info(
+                f"Finished creating derivative images for master file {master_image.file.name}."
+            )
+
+    except (CommandError, ValueError) as ex:
+        logger.error(ex)
+        raise
+    finally:
+        # Delete the temporary files;
+        # don't throw FileNotFoundError if for some reason it doesn't exist.
+        try:
+            Path(submaster_file_name).unlink(missing_ok=True)
+            Path(thumbnail_file_name).unlink(missing_ok=True)
+        except UnboundLocalError:
+            # Swallow this, which happens when derivative creation fails
+            # so file_name variable is not defined.
+            pass
+
+
+def delete_existing_derivative_images(master_images: QuerySet) -> None:
+    """Delete existing derivative images from masters."""
+    try:
+        for master_image in master_images:
+            project_item = master_image.item
+            logger.info(
+                f"Deleting existing derivative images for project item {project_item}."
```

Comment:
Please add the project item id to this message, for easier lookup / troubleshooting if needed - at present it just has the item title.  Something like "...for project item {project_item} ({project_item.id})" should be fine.